### PR TITLE
Fix scalap parsing/printing of enum and class constant types

### PR DESCRIPTION
**Pull request by retronym**
Fixes scala/bug#5194

### CODE REVIEW

File: test/files/scalap/constants.check [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  final val ShortConstant: scala.Short = { /* compiled code */ }
+  final val IntConstant: scala.Int = { /* compiled code */ }
+  final val LongConstant: scala.Long = { /* compiled code */ }
+  final val FloatConstant: scala.Float = { /* compiled code */ }
+  final val DoubleConstant: scala.Double = { /* compiled code */ }
+  final val NullConstant: scala.Null = { /* compiled code */ }
+  final val ClassConstant: java.lang.Class[scala.Predef.String] = { /* compiled code */ }
+  final val ClassConstant2: java.lang.Class[scala.Some[_]] = { /* compiled code */ }
+  final val EnumConstant: java.util.concurrent.TimeUnit = { /* compiled code */ }
+  final val StringConstant: java.lang.Class[scala.Predef.String] = { /* compiled code */ }
```
Comment (retronym) [inline test/files/scalap/constants.check:15]: Strictly speaking, scalap should render these methods as `final val IntConstant = 1`, rather than with the widened type. But I'm only here on a drive-by fix and just want to avoid the crash that stops `jardiff` from comparing the signatures in `scala/concurrent/duration/package.class`.

File: src/scalap/scala/tools/scalap/scalax/rules/scalasig/ScalaSig.scala [hunk 2]
```diff
@@ -246,7 +246,9 @@ object ScalaSigEntryParsers extends RulesWithState with MemoisableRules {
       32 -~ longValue ^^ (java.lang.Double.longBitsToDouble),
       33 -~ nameRef,
       34 -^ null,
-      35 -~ typeRef)
+      35 -~ typeRef,
+      36 -~ symbolRef
```
Comment (retronym) [inline src/scalap/scala/tools/scalap/scalax/rules/scalasig/ScalaSig.scala:250]: Review by @lrytz, this is 2009 [calling](https://github.com/scala/scala/commit/c762ae353b9ff8beec6ed9516b70134e1f04534d#diff-de53eb5601161b428c5d8a57b0a81737R57) :)

Comment (lrytz) [reply src/scalap/scala/tools/scalap/scalax/rules/scalasig/ScalaSig.scala:250]: > more annotations cleanup.

Great commit message for a +159 -194 commit 😂

**Review by lrytz — APPROVED** [review]
